pkg/kubelet/nodestatus/setters.go

```diff
@@ -612,6 +612,10 @@ func getCurrentRuntimeReadiness(runtimeCondition *v1.NodeCondition, workloadType
 	statusSet := false
 	for _, status := range runtimeServiceStatus {
 		if status == true {
+			if runtimeCondition.Status == v1.ConditionTrue {
```

Comment:
This will work by not changing anything in runtimeCondition. But maybe it's better to do it outside of this function, and this function still returns latest status/reason/lastTransactionTime.  if we look at other similar conditions (memoryPressure, diskPressure, PIDPressure, etc), they all check and decide if to update conditions in setters, like functions MemoryPressureCondition(), DiskPressureCondition).  So I think adding comparisons there will make the comparison logic explicit and easy to understand, and also make styles consistent.

Reply:
it is reasonable to follow the convention and keep code consistent. will change accordingly.